jujugui/static/gui/src/app/components/env-switcher/list/list.js

```diff
@@ -47,18 +50,25 @@ YUI.add('env-list', function() {
     generateModelList: function() {
       const models = this.state.envs;
       if (!models.length) {
-        return (
-          <li className="env-list__environment" key="none">
-            No models available, click below to view your profile and create a
-            new model.
-          </li>
-        );
+        return false;
       }
       const auth = this.props.authDetails;
       const currentUser = auth ? auth.user : null;
-      return models.map(model => {
+      const modelsWithoutController = models.sort((a, b) => {
```

Comment:
I see the code, but I have no idea why you're doing what you're doing here. Can you add some comments to this effect? And why would a model not have a controller?